sympy/polys/matrices/domainmatrix.py

```diff
@@ -462,36 +463,34 @@ def to_dense(self):
 
         return self.from_rep(SDM.to_ddm(self.rep))
 
-    def _unify_domain(self, other):
-        """Convert self and other to a common domain"""
-        K1 = self.domain
-        K2 = other.domain
-        if K1 == K2:
-            return self, other
-        K = K1.unify(K2)
-        if K1 != K:
-            self = self.convert_to(K)
-        if K2 != K:
-            other = other.convert_to(K)
-        return self, other
-
-    def _unify_fmt(self, other, fmt):
-        """Convert self and other to the same format.
-
-        If both are sparse or both are dense then return both unmodified.
+    @classmethod
+    def _unify_domain(cls, *matrices):
+        """Convert matrices to a common domain"""
+        domains = [matrix.domain for matrix in matrices]
+        if len(set(domains)) == 1:
+            return matrices
+        domain = reduce(lambda x, y: x.unify(y), domains)
```

Comment:
It would be better to use `set(domains)` here.

Reply:
I wonder if there are any reasonable uses where an empty list of matrices would be given here and a default domain such as ZZ could be specified.

Reply:
As long as `Domain.unify` can't handle nullary union, I think that it can be left as exception rather than defining `ZZ`.
I haven't decided that `ZZ` could be the natural answer for that, I rather thought about `EmptyDomain`, but I'm not sure if that's mathematical.
I think that we may also extend `Domain.unify` itself to work with n-ary arguments, but I'm afraid that this can be complicated

Reply:
> It would be better to use `set(domains)` here.

I'm concerned about whether the `Domain.union` is guaranteed to give unique result regardless of the permutation of the arguments. If it is not the case, I'm afraid that using `set` directly can give indeterministic result.

Reply:
I meant that the default could be give like:
```
K = cls._unify_domain(matrices, ZZ)
```
but maybe it's best just to leave that until there's a use for it.

If `unify` works then I think it should be expected to give the same result regardless of the order of the domains.